pkgs/dartpad_shared/lib/headers.dart

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+const _loggingOnHeaderName = 'X-Logging-On';
+
+/// A class that represents the custom request headers for DartPad.
+///
+/// Standard serialization is not used, because the header name should have
+/// prefix 'X-'.
+class DartPadRequestHeaders {
+  static final instance = DartPadRequestHeaders();
+
+  /// The default value is true.
+  ///
+  /// The value is set to false for clients where debug asserts are enabled.
+  /// It can be set to false in constructor.
+  late final bool loggingOn;
+
+  DartPadRequestHeaders({bool? loggingOn}) {
+    if (loggingOn != null) {
+      this.loggingOn = loggingOn;
+      return;
+    }
+
+    var setLoggingOn = true;
+    assert(() {
```

Comment:
What's the purpose of turning logging off when running in debug mode?

Reply:
It was your request to turn off logging for probes, right? So, if we turn off logging for ui in debug mode, propes and other dev related calls will not be logged.

Another option is to add a label to such logs.

Did I miss something?

Reply:
That's correct, I think I understand why this is written this way. I just wanted to make sure that we are still able to log requests when I run dart_services locally. This class is defined in dartpad_shared, but as long as it's only used in dartpad_ui this should be okay.

Maybe this class should be defined in dartpad_ui to avoid confusion?

Reply:
I was defining it in shared because it is supposed to be encoded in ui and decoded in backend. 
But since we are not going to use testWidgets with probes, we do not need it in UI at all.